bot/exts/utils/snekbox/_eval.py

```diff
@@ -0,0 +1,179 @@
+        if self.returncode is None:
+            error = self.stdout.strip()
+        elif self.returncode == 255:
+            error = "A fatal NsJail error occurred"
+        return error
+
+    @property
+    def files_error_message(self) -> str:
+        """Return an error message corresponding to the failed files."""
+        if not self.failed_files:
+            return ""
+
+        failed_files = f"({self.get_failed_files_str()})"
+
+        n_failed = len(self.failed_files)
+        s_upload = "uploads" if n_failed > 1 else "upload"
+
+        msg = f"{Emojis.failed_file} {n_failed} file {s_upload} {failed_files} failed"
+
+        # Exceeded file count limit
+        if (n_failed + len(self.files)) > FILE_COUNT_LIMIT:
+            s_it = "they" if n_failed > 1 else "it"
+            msg += f" as {s_it} exceeded the {FILE_COUNT_LIMIT} file limit."
+        # Exceeded file size limit
+        else:
+            s_each_file = "each file's" if n_failed > 1 else "its file"
+            msg += f" because {s_each_file} size exceeds {sizeof_fmt(FILE_SIZE_LIMIT)}."
+
+        return msg
+
+    def get_failed_files_str(self, char_max: int = 85) -> str:
+        """
+        Return a string containing the names of failed files, truncated char_max.
+
+        Will truncate on whole file names if less than 3 characters remaining.
+        """
+        names = []
+        for file in self.failed_files:
+            # Only attempt to truncate name if more than 3 chars remaining
+            if char_max < 3:
+                names.append("...")
+                break
+
+            to_display = min(char_max, len(file))
+            name_short = file[:to_display]
+            # Add ellipsis if name was truncated
+            if to_display < len(file):
+                name_short += "..."
+                names.append(name_short)
+                break
+
+            char_max -= len(file)
+            names.append(file)
+
+        text = ", ".join(names)
+        # Since the file names are provided by user
+        text = escape_markdown(text)
+        text = escape_mentions(text)
+        return text
+
+    def get_message(self, job: EvalJob) -> str:
+        """Return a user-friendly message corresponding to the process's return code."""
+        msg = f"Your {job.version} {job.name} job"
+
+        if self.returncode is None:
+            msg += " has failed"
+        elif self.returncode == 128 + SIGKILL:
+            msg += " timed out or ran out of memory"
+        elif self.returncode == 255:
+            msg += " has failed"
+        else:
+            msg += f" has completed with return code {self.returncode}"
+            # Try to append signal's name if one exists
+            with contextlib.suppress(ValueError):
+                name = Signals(self.returncode - 128).name
+                msg += f" ({name})"
+
+        return msg
+
+    @classmethod
+    def from_dict(cls, data: dict[str, str | int | list[dict[str, str]]]) -> EvalResult:
```

Comment:
I don't think it's really needed here since the only usage of this is 1 line at 
```py
return EvalResult.from_dict(await resp.json())
```
where resp.json() wouldn't be typed anyways. If we weren't using the dataclass in internal API and passing the dict around instead then it might be more useful to have a TypedDict.